src/redux/features/users/userSlice.ts

```diff
@@ -18,16 +18,52 @@ export const getAllUsers = createAsyncThunk(
   },
 );
 
+export const getAllProjects = createAsyncThunk(
+  'All/projects',
+  async () => {
+    // eslint-disable-next-line no-useless-catch
+    try{
+      const { data } = await API.get('/projects');
+      return data;
+    }catch(error: any){
+      throw error?.response?.data;
+    }
+   
+  },
+);
+
+export const getAllUserByProjects = createAsyncThunk(
+  'AllUsers/projects',
+  async (projectId: number) => {
+    // eslint-disable-next-line no-useless-catch
+    try{
+      const { data } = await API.get(`/projects/${projectId}/users`);
+      return data;
+    }catch(error: any){
+      throw error?.response?.data;
+    }
+   
+  },
+);
+
 interface InitialState {
   users: User[];
+  usersByProject: User[];
+  projects:any [],
```

Comment:
@bizip Avoid using `any` type wherever possible, so define the project type and use it here instead